### PR TITLE
Feat: added Alert component + improved location page

### PR DESCRIPTION
**Pull request by robinpiets**
## Updates:
- added `Alert` component to messages and alerts
- fixed Toevoeging [suffix] input, by always showing it
- refactored code
- updated text

---

### GraphQL error:

image

---
### Location error:
image

---
### Location found:
image

### CODE REVIEW

File: apps/client/src/components/Location/LocationFinder.js [hunk 1]
```diff
@@ -96,63 +96,75 @@ const LocationFinder = (props) => {
...
+          content="De resultaten worden ingeladen."
+        />
       )}
 
-      {postalCode && houseNumberFull && addressMatches && (
-        <>
+      {notFoundAddress && !queryError && (
+        <StyledAlert
+          heading="Helaas. Wij kunnen geen adres vinden bij uw postcode en huisnummer combinatie. "
+          style={{
```
Comment (svenjens) [inline apps/client/src/components/Location/LocationFinder.js:138]: Styled alert only has a margin-bottom: 25px prop, maybe we should handle the styling different here?

File: apps/client/src/pages/ErrorPage.js [hunk 2]
```diff
@@ -9,7 +9,7 @@ const ErrorPage = ({ error }) => (
     <Helmet>
       <title>Er is een fout opgetreden - Amsterdam Vergunningchecker</title>
     </Helmet>
-    <Error error={error} />
+    <Error error={error} content={error.message} />
```
Comment (svenjens) [inline apps/client/src/pages/ErrorPage.js:12]: Feels weird to split error and content, when error already has the content. 
We send error to show us the error stack as developers right? 

And this is the only place we send content? as prop.

File: apps/client/src/components/Location/LocationFinder.js [hunk 3]
```diff
@@ -96,63 +96,75 @@ const LocationFinder = (props) => {
...
+            backgroundColor: "white",
```
Comment (svenjens) [inline apps/client/src/components/Location/LocationFinder.js:139]: StyledAlert only has marginBottom 25px. Maybe we should handle styling different here?

File: apps/client/src/components/Location/LocationFinder.js (hunk 1 above)
Comment (robinpiets) [reply apps/client/src/components/Location/LocationFinder.js:138]: Yeah, I was thinking the same thing, just didn't have a proper solution yet. You got a suggestion?

File: apps/client/src/pages/ErrorPage.js (hunk 2 above)
Comment (robinpiets) [reply apps/client/src/pages/ErrorPage.js:12]: Good point, I will refactor this

File: apps/client/src/components/Location/LocationFinder.js (hunk 1 above)
Comment (robinpiets) [reply apps/client/src/components/Location/LocationFinder.js:138]: This should solve it: #344

File: apps/client/src/components/Location/LocationFinder.js (hunk 3 above)
Comment (robinpiets) [reply apps/client/src/components/Location/LocationFinder.js:139]: This should solve it: #344

File: apps/client/src/components/Location/LocationFinder.js [hunk 4]
```diff
@@ -16,10 +15,10 @@ const LocationFinder = (props) => {
   const [houseNumber, setHouseNumber] = useState(props.houseNumber);
   const [houseNumberFull, setHouseNumberFull] = useState(props.houseNumberFull);
   const [touched, setTouched] = useState({});
-  const { setAddress } = props;
+  const { setAddress, setQueryError } = props;
```
Comment (afjlambert) [inline apps/client/src/components/Location/LocationFinder.js:18]: Can we maybe rename to `onError` or `handleError`? That would be more in line with with other components imo.

File: apps/client/src/components/Location/LocationFinder.js [hunk 5]
```diff
@@ -96,63 +96,75 @@ const LocationFinder = (props) => {
...
-        }
+        errorMessage={validate("houseNumberFull", houseNumberFull, true)}
       />
 
-      {notFoundAddress && (
-        <StyledErrorParagraph>
-          Er is geen adres in Amsterdam gevonden op basis van deze gegevens.
-          Probeer het opnieuw.
-        </StyledErrorParagraph>
+      <>
```
Comment (afjlambert) [inline apps/client/src/components/Location/LocationFinder.js:102]: Why do you add this fragment here?

Comment (afjlambert) [reply apps/client/src/components/Location/LocationFinder.js:102]: I guess this part will be changed to `<input type='text'` soon.

File: apps/client/src/intros/DakkapelIntro.js [hunk 6]
```diff
@@ -37,12 +37,14 @@ export default () => (
     <List variant="bullet">
       <ListItem>
         Wilt u de dakkapel plaatsen op een woonwagen, een tijdelijk gebouw, een
-        blokhut of een vakantiehuis? Bel dan de gemeente op 14 020, maandag tot
-        en met vrijdag van 08.00 uur tot 18.00 uur
+        blokhut of een vakantiehuis? Bel dan de gemeente op{" "}
+        <a href="tel:14020">14 020</a>, maandag tot en met vrijdag van 08.00 uur
```
Comment (afjlambert) [inline apps/client/src/intros/DakkapelIntro.js:41]: :thumbsup:

File: apps/client/src/pages/LocationPage.js [hunk 7]
```diff
@@ -48,11 +50,24 @@ const LocationPage = ({ topic }) => {
       <Helmet>
         <title>Invullen adres - {text.heading}</title>
       </Helmet>
+      {queryError && (
+        <Error
+          heading="Helaas. Wij kunnen nu geen locatiegegevens opvragen waardoor u deze
+        check niet kunt gebruiken."
```
Comment (afjlambert) [inline apps/client/src/pages/LocationPage.js:56]: I think we use backtick's for multiline strings 🤔

File: apps/client/src/components/Error.js [hunk 8]
```diff
@@ -1,11 +1,22 @@
 import React from "react";
-import { Heading, Paragraph } from "@datapunt/asc-ui";
+// Component Alert has wrong export name `AlertMessage`. When the new update is released, just remove `AlertMessage as`
+import { AlertMessage as Alert } from "@datapunt/asc-ui";
 
-const Error = ({ error }) => (
+const Error = ({ error, heading, content, children }) => (
   <>
```
[the remark this answers is not in the document]
Comment (robinpiets) [reply apps/client/src/components/Error.js:6]: It's gone. I don't know why you still see it there?

image

File: apps/client/src/pages/LocationPage.js (hunk 7 above)
Comment (robinpiets) [reply apps/client/src/pages/LocationPage.js:56]: Good catch! 🎣

File: apps/client/src/components/Location/LocationFinder.js (hunk 5 above)
Comment (robinpiets) [reply apps/client/src/components/Location/LocationFinder.js:102]: Another good catch! 🎣

Comment (robinpiets) [reply apps/client/src/components/Location/LocationFinder.js:102]: `<input` is going to be completely rewritten in `asc-ui`, better wait for that and see what comes out

File: apps/client/src/components/Location/LocationFinder.js (hunk 4 above)
Comment (robinpiets) [reply apps/client/src/components/Location/LocationFinder.js:18]: I understand the name not proper, however, It's a function to invoke `setState` so `onError` or `handleError` would  also not be correct. I will work this out

Comment (robinpiets) [reply apps/client/src/components/Location/LocationFinder.js]: How about this @afjlambert? `setErrorMessage(graphqlError);`

File: apps/client/src/components/Error.js (hunk 8 above)
[the remark this answers is not in the document]
Comment (afjlambert) [reply apps/client/src/components/Error.js:6]: I mean you do `<><Alert/></>` but you can also just `<Alert />` right? Not sure why prettier doesn't fix this... 🤔

[the remark this answers is not in the document]
Comment (robinpiets) [reply apps/client/src/components/Error.js]: Removed it

**Pull request merged**